### ISSUE TITLE
integration tests for testing uses of slang as a library

### ISSUE THREAD

**Issue by expipiplus1** [shader-slang/slang#5537]
For example:

- Building slang as part of a higher-level cmake project
- Downloading and testing releases
- Using slang as a submodule

**Comment by bmillsNV** [comment]
@expipiplus1 we think our testing has stabilized a bit. Are you OK if we close this?

**Comment by expipiplus1** [comment]
I'm happy to close it if we think the testing covers these use cases. But to my knowledge we still don't do any testing of generated releases ourselves (although they're definitely tested as part of all our consumers' workflows, maybe that's sufficient). Happy to be corrected if we actually do do this though

**Comment by aasgaonkar** [comment]
@expipiplus1 We have ongoing effort to run rhi and slangpy tests in slang CI. would that be considered as integration tests? they will consume the build from slang CI and run their tests.

**Comment by expipiplus1** [comment]
Yeah, these are definitely good things to have, but I think it would still be a good idea to do basic tests of how we expect people to consume slang

**Comment by aasgaonkar** [comment]
agree. I will keep this issue active to track that kind of testing. not sure if we have something to add, but we should if not.

**Comment by aasgaonkar** [comment]
removing this from parent task for current CI improvements.
